src/absil/ilwritepdb.fs

```diff
@@ -169,28 +189,52 @@ let pdbGetPdbDebugInfo (embeddedPDBChunk: BinaryChunk) (uncompressedLength: int6
       iddType = 17                                                              // IMAGE_DEBUG_TYPE_EMBEDDEDPDB
       iddTimestamp = 0
       iddData = iddPdbBuffer                                                    // Path name to the pdb file when built
-      iddChunk = embeddedPDBChunk
+      iddChunk = embeddedPdbChunk
     }
 
-let pdbGetDebugInfo (mvid: byte[]) (timestamp: int32) (filepath: string) (cvChunk: BinaryChunk) (embeddedPDBChunk: BinaryChunk option) (uncompressedLength: int64) (stream: MemoryStream option) = 
-    match stream, embeddedPDBChunk with
-    | None, _  | _, None -> [| pdbGetCvDebugInfo mvid timestamp filepath cvChunk |]
-    | Some s, Some chunk -> [| pdbGetCvDebugInfo mvid timestamp filepath cvChunk; pdbGetPdbDebugInfo chunk uncompressedLength s |]
+let pdbChecksumDebugInfo timestamp (checksumPdbChunk: BinaryChunk) (algorithmName:string) (checksum: byte[]) =
+    let iddBuffer =
+        let alg = Encoding.UTF8.GetBytes(algorithmName)
+        let buffer = Array.zeroCreate (alg.Length + 1 + checksum.Length)
+        Buffer.BlockCopy(alg, 0, buffer, 0, alg.Length)
+        Buffer.BlockCopy(checksum, 0, buffer, alg.Length + 1, checksum.Length)
+        buffer
+    { iddCharacteristics = 0                                                    // Reserved
+      iddMajorVersion = 1                                                       // VersionMajor should be 1
+      iddMinorVersion = 0x0100                                                  // VersionMinor should be 0x0100
+      iddType = 19                                                              // IMAGE_DEBUG_TYPE_CHECKSUMPDB
+      iddTimestamp = timestamp
+      iddData = iddBuffer                                                       // Path name to the pdb file when built
+      iddChunk = checksumPdbChunk
+    }
 
-// Document checksum algorithms
-let guidSourceHashMD5 = System.Guid(0x406ea660u, 0x64cfus, 0x4c82us, 0xb6uy, 0xf0uy, 0x42uy, 0xd4uy, 0x81uy, 0x72uy, 0xa7uy, 0x99uy) //406ea660-64cf-4c82-b6f0-42d48172a799
-let hashSizeOfMD5 = 16
+let pdbGetPdbDebugDeterministicInfo (deterministicPdbChunk: BinaryChunk) =
+    { iddCharacteristics = 0                                                    // Reserved
+      iddMajorVersion = 0                                                       // VersionMajor should be 0
+      iddMinorVersion = 0                                                       // VersionMinor should be 0x0100
```

Comment:
Comment and data don't agree.

Reply:
> Comment and data don't agree.

Indeed they don't code is right, comment is copy/paste error.

Thx.